main.rb

```diff
@@ -0,0 +1,144 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each { |n| puts "#{n}!" }
+
+  def my_each_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each_each_with_index { |_n, x| print x }
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = to_a
+    final_array = []
+    arr.my_each { |x| final_array.push(x) if yield(x) }
+    final_array
+  end
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].my_select { |x| puts x if x.even? }
+
+  def my_all?(arg = nil)
+    arr = to_a
+    if block_given?
+      arr.my_each { |x| return true unless yield(x) == false }
+      return true
+    elsif arg.nil?
+      return false
+    elsif !arg.nil? && (arg.is_a? Regexp)
+      arr.my_each { |x| return false unless x.match(arg) }
+    elsif (arg.is_a? Class) && !arg.nil?
+      arr.my_each { |x| return false unless [x.class, x.class.superclass].include?(arg) }
+    else
+      arr.my_each { |i| return false if i != arg }
+
+    end
+    true
+  end
```

Comment:
This method throws the following errors:

```
  1) enumerables #my_all? returns false if the block returns false or nil
     Failure/Error: expect(array.my_all?(&false_block)).to eq(array.all?(&false_block))
     
       expected: false
            got: true
     
       (compared using ==)
     
       Diff:
       @@ -1 +1 @@
       -false
       +true
       
     # ./spec/enumerables_spec.rb:146:in `block (3 levels) in <top (required)>'

  2) enumerables #my_all? returns false if the block returns false or nil::range
     Failure/Error: expect(range.my_all?(&false_block)).to eq(range.all?(&false_block))
     
       expected: false
            got: true
     
       (compared using ==)
     
       Diff:
       @@ -1 +1 @@
       -false
       +true
       
     # ./spec/enumerables_spec.rb:150:in `block (3 levels) in <top (required)>'

  3) enumerables #my_all? when no block or argument is given returns true when none of the collection members are false or nil
     Failure/Error: expect(true_array.my_all?).to be true_array.all?
     
       expected true
            got false
     # ./spec/enumerables_spec.rb:162:in `block (4 levels) in <top (required)>'
```

Please test this method so that it produces the following results:

```
array = [2, 4, 3, 1]
rante = (1..5)
false_block = proc { |num| num > 7 }
true_array = [1, true, 'hi', []]

array.my_all?(&false_block)
range.my_all?(&false_block)
# These two should be false

true_array.my_all?
# This should be true
```